pkg/api/routes.go

```diff
@@ -659,9 +742,17 @@ func (rh *RouteHandler) GetBlob(response http.ResponseWriter, request *http.Requ
 	defer repo.Close()
 
 	response.Header().Set("Content-Length", fmt.Sprintf("%d", blen))
-	response.Header().Set(constants.DistContentDigestKey, digest)
+
+	status := http.StatusOK
+
+	if partial {
+		status = http.StatusPartialContent
```

Comment:
The `Content-Range` header must be set (see: https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests#single_part_ranges)

`bsize`is the total blob size returned as an extra return value from `imgStore.GetBlobPartial`
`(from+blen-1)` is used so that we limit the `to` value to the maximum blob size

```suggestion
		status = http.StatusPartialContent
		response.Header().Set("Content-Range", fmt.Sprintf("bytes %d-%d/%d", from, (from+blen-1), bsize))
```